### PR TITLE
Improve LastRemovalEventId determinism by adding ordering

### PR DESCRIPTION
**Pull request by quantumagi**
This PR adds an `OrderByDescending` to the `FirstOrDefault` used to determine the last kick poll.

Also, two methods are added to unify two slightly different implementations of the same logic.

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.PoA/Voting/JoinFederationRequestService.cs [hunk 1]
```diff
@@ -49,6 +49,23 @@ public JoinFederationRequestService(ICounterChainSettings counterChainSettings,
...
+            Script collateralScript = PayToPubkeyHashTemplate.Instance.GenerateScriptPubKey(joinRequest.CollateralMainchainAddress);
+            BitcoinAddress bitcoinAddress = collateralScript.GetDestinationAddress(counterChainNetwork);
+            var collateralFederationMember = new CollateralFederationMember(joinRequest.PubKey, false, joinRequest.CollateralAmount, bitcoinAddress.ToString());
+
+            return (network.Consensus.ConsensusFactory as CollateralPoAConsensusFactory).SerializeFederationMember(collateralFederationMember);
+        }
+
+        public static void SetLastRemovalEventId(JoinFederationRequest joinRequest, byte[] federationMemberBytes, VotingManager votingManager)
+        {
+            Poll poll = votingManager.GetApprovedPolls().OrderByDescending(p => p.PollStartBlockData.Height).FirstOrDefault(x => x.IsExecuted &&
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.PoA/Voting/JoinFederationRequestService.cs:63]: You can use `this.votingManager.GetExecutedPolls()` here rather :)

File: src/Stratis.Features.Collateral/JoinFederationRequestMonitor.cs [hunk 2]
```diff
@@ -41,8 +37,6 @@ public JoinFederationRequestMonitor(VotingManager votingManager, Network network
 
         public Task InitializeAsync()
         {
-            this.blockConnectedToken = this.signals.Subscribe<BlockConnected>(this.OnBlockConnected);
```
Comment (fassadlr) [inline src/Stratis.Features.Collateral/JoinFederationRequestMonitor.cs:44]: I dont think this should be removed as we still depend on the block connected event to call `OnBlockConnected`?

Comment (quantumagi) [reply src/Stratis.Features.Collateral/JoinFederationRequestMonitor.cs]: Lol Wtf. Right you are.

File: src/Stratis.Bitcoin.Features.PoA/Voting/JoinFederationRequestService.cs (hunk 1 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.PoA/Voting/JoinFederationRequestService.cs]: Changed.

**Review by fassadlr — APPROVED** [review]

**Pull request merged**